pkg/scheduler/framework/v1alpha1/waiting_pods_map.go

```diff
@@ -27,19 +27,19 @@ import (
 
 // waitingPodsMap a thread-safe map used to maintain pods waiting in the permit phase.
 type waitingPodsMap struct {
-	pods map[types.UID]WaitingPod
+	pods map[types.UID]*waitingPod
```

Comment:
Why make this change? for directly accessing `waitingPod.s`? If so, we can add a function to WaitingPod interface:

```go
type WaitingPod interface {
    StatusChan() <-chan *Status
}
```

Reply:
Main reason is that if we expose channel then we risk that custom plugin code will use `wp.StatusChan()` and intercept status that was meant for `WaitOnPermit()` and `WaitOnPermit()` will block forever. 

One option could be `Wait() *Status` or `Wait() (allowed bool)` which blocks until waiting pod is rejected or allowed and always return the same value that was memorized, however it would require more complicated `*waitingPod` implementation.

I decided to simply use `waitingPod` inside `waitingPodMap` given that both types are internal and we do not need to use different implementation of the `WaitingPod` inside the `waitingPodMap`.

Reply:
Using a concrete value here (instead of an interface) would lose the flexibility of building tests, but it's affordable. So SGTM.